File: DevExpress-Demo/VB/DiagramDemo.Wpf/Helpers/OrgChartHelpers.vb
Imports DevExpress.Diagram.Core
Imports DevExpress.Xpf.Diagram
Imports System
Imports System.Linq
Imports System.Runtime.CompilerServices

Namespace DevExpress.Diagram.Demos

    Public Module OrgChartHelpers

        Private ReadOnly styles As DiagramItemStyleId() = DiagramShapeStyleId.Styles.ToArray()

        Public Function GetStyleID(ByVal item As DiagramItem) As DiagramItemStyleId
            Return GetStyle(GetItemLevel(item))
        End Function

        Public Function GetStyle(ByVal id As Integer) As DiagramItemStyleId
            Select Case id Mod 7
                Case 0
                    Return DiagramShapeStyleId.Balanced1
                Case 1
                    Return DiagramShapeStyleId.Balanced2
                Case 2
                    Return DiagramShapeStyleId.Balanced3
                Case 3
                    Return DiagramShapeStyleId.Balanced4
                Case 4
                    Return DiagramShapeStyleId.Balanced5
                Case 5
                    Return DiagramShapeStyleId.Balanced6
                Case 6
                    Return DiagramShapeStyleId.Balanced7
            End Select

            Return DiagramShapeStyleId.Variant1
        End Function

        <Extension()>
        Public Function GetBrightStyle(ByVal styleId As DiagramItemStyleId) As DiagramItemStyleId
            Dim index = Array.IndexOf(styles, styleId)
            If index = -1 Then Return DiagramShapeStyleId.Variant1
            Return styles(If(index > 10, index - 7, index))
        End Function

        <Extension()>
        Public Function GetPaleStyle(ByVal styleId As DiagramItemStyleId) As DiagramItemStyleId
            Dim index = Array.IndexOf(styles, styleId)
            If index = -1 Then Return DiagramShapeStyleId.Variant1
            Return styles(If(index < 39, index + 7, index))
        End Function

        Public Function GetItemLevel(ByVal item As DiagramItem) As Integer
            Dim level = 0
            Dim parent = GetParent(item)
            While parent IsNot Nothing
                level += 1
                parent = GetParent(parent)
            End While

            Return level
        End Function

        Private Function GetParent(ByVal item As DiagramItem) As DiagramItem
            Dim incomingConnector = item.IncomingConnectors.FirstOrDefault()
            Return If(incomingConnector Is Nothing, Nothing, CType(incomingConnector.BeginItem, DiagramItem))
        End Function
    End Module
End Namespace

File: DevExpress-Demo/CS/DiagramDemo.Wpf/Helpers/OrgChartHelpers.cs
using DevExpress.Diagram.Core;
using DevExpress.Xpf.Diagram;
using System;
using System.Windows;
using System.Linq;
using DevExpress.Mvvm.Native;
using DevExpress.Diagram.Core.Native;

namespace DevExpress.Diagram.Demos {
    public static class OrgChartHelpers {
        readonly static DiagramItemStyleId[] styles = DiagramShapeStyleId.Styles.ToArray();
        public static DiagramItemStyleId GetStyleID(DiagramItem item) {
            return GetStyle(GetItemLevel(item));
        }
        public static DiagramItemStyleId GetStyle(int id) {
            switch(id % 7) {
                case 0:
                    return DiagramShapeStyleId.Balanced1;
                case 1:
                    return DiagramShapeStyleId.Balanced2;
                case 2:
                    return DiagramShapeStyleId.Balanced3;
                case 3:
                    return DiagramShapeStyleId.Balanced4;
                case 4:
                    return DiagramShapeStyleId.Balanced5;
                case 5:
                    return DiagramShapeStyleId.Balanced6;
                case 6:
                    return DiagramShapeStyleId.Balanced7;
            }
            return DiagramShapeStyleId.Variant1;
        }
        public static DiagramItemStyleId GetBrightStyle(this DiagramItemStyleId styleId) {
            var index = Array.IndexOf(styles, styleId);
            if(index == -1)
                return DiagramShapeStyleId.Variant1;
            return styles[index > 10 ? index - 7 : index];
        }
        public static DiagramItemStyleId GetPaleStyle(this DiagramItemStyleId styleId) {
            var index = Array.IndexOf(styles, styleId);
            if(index == -1)
                return DiagramShapeStyleId.Variant1;
            return styles[index < 39 ? index + 7 : index];
        }
        public static int GetItemLevel(DiagramItem item) {
            var level = 0;
            var parent = GetParent(item);
            while(parent != null) {
                level++;
                parent = GetParent(parent);
            }
            return level;
        }
        static DiagramItem GetParent(DiagramItem item) {
            var incomingConnector = item.IncomingConnectors.FirstOrDefault();
            return incomingConnector == null ? null : (DiagramItem)incomingConnector.BeginItem;
        }
    }
}
